Document/0x05j-Testing-Resiliency-Against-Reverse-Engineering.md

```diff
@@ -40,6 +43,32 @@ The following is a sample attestation result:
 }
 ```
 
+**ctsProfileMatch Vs basicIntegrity**
+
+The SafetyNet Attestation API initially provided a single value called basicIntegrity to help developers determine the integrity of a device. As the API evolved, Google introduced a new, stricter check whose results appear in a value called ctsProfileMatch, which allows developers to more finely evaluate the devices on which their app is running.
+
+In broad terms, basicIntegrity gives you a signal about the general integrity of the device and its API. Rooted devices fail basicIntegrity, as do emulators, virtual devices, and devices with signs of tampering, such as API hooks.
+
+On the other hand, ctsProfileMatch gives you a much stricter signal about the compatibility of the device. Only unmodified devices that have been certified by Google can pass ctsProfileMatch. Devices that will fail ctsProfileMatch include the following:
+
+- Devices that fail basicIntegrity
+- Devices with an unlocked bootloader
+- Devices with a custom system image (custom ROM)
+- Devices for which the manufactured didn't apply for, or pass, Google certification
+- Devices with a system image built directly from the Android Open Source Program source files
+- Devices with a system image distributed as part of a beta or developer preview program (including the Android Beta Program)
+
+**Do's and Don'ts of** `SafetyNetApi.attest`
+
+-  Create a large (16 bytes or longer) random number on your server using a cryptographically-secure random function so that a malicious user can not reuse a successful attestation result in place of an unsuccessful result
+- Trust APK information (apkPackageName, apkCertificateDigestSha256 & apkDigestSha256) only if the value of ctsProfileMatch is true.
+- Using a secure connection, entire JWS response should be send to your server, for verification. It isn't recommend to perform the verification directly in the app because, in that case, there is no guarantee that the verification logic itself hasn't been modified.
+- The verify() method only validates that the JWS message was signed by SafetyNet. It doesn't verify that the payload of the verdict matches your expectations. Also, it has a fixed rate limit of 10,000 requests per day, per project. Hence, use the verify() method only for testing during the initial development stage and not in a production scenario.
+- The SafetyNet Attestation API gives you snapshot of the state of a device at the moment when the attestation request was made. A successful attestation doesn't necessarily mean that the device would have passed attestation in the past, or that it will in the future. It's recommend to plan a strategy to use the least amount of attestations required to satisfy the use case.
+
+
+Follow this [checklist](https://developer.android.com/training/safetynet/attestation-checklis) to ensure that you've completed each of the steps needed to integrate the SafetyNet Attestation API into the app.

```

Comment:
Can you please reformat the url as [checklist](https://developer.android.com/training/safetynet/attestation-checklis "attestation checklist") ?